### PR TITLE
CMake: Add LLVM build for Windows and other linker fixes

### PR DESCRIPTION
**Pull request by alef**
#### Summary
Build "CMake: Add LLVM build for Windows and other linker fixes"

#### Purpose of change
* Compile and link using LLVM v17 for Windows.
* Simplify logic for embedding the manifest.
* Fix MINGW executable not running right after the build.
* Fix `LNK4098` from MSVC

#### Describe the solution
* Add a new CMake preset for using LLVM on Windows. Requires `LLVM_ROOT` to be defined. Default `C:\Program Files\LLVM`
* LLVM build uses `clang-cl.exe` so that we can reuse flags from `MSVC.cmake` toolchain file
* Removed a weird logic to handle manifest because it wasn't clear `/MANIFEST:NO` was about commanding the linker to **not automatically** generate and embed a manifest.
* Instead of manually stripping the MSYS/MINGW build manually, just add `--strip-debug` to `ld.exe` to make the executable run on native Windows.
* Handle `/NODEFAULTLIB:` and `CMAKE_MSVC_RUNTIME_LIBRARY` to fix MSVC's linker warning `LNK4098` about conflicting libraries.
* Do not always write `version.h` when configuring the build. Avoid `version.cpp` to be rebuilt when there is no need.

#### Describe alternatives you've considered
Split this PR in smaller ones.

#### Testing
✅ Build and run LLVM, MSVC and UCRT64 executables

#### Additional context
N/A

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -48,13 +48,13 @@ if (TILES)
                 ${MESSAGES_CPP}
                 ${RESOURCE_RC})
         if(MSYS2)
-            # /MANIFESTINPUT not supported by GNU linker
-            # Defaul embed via `windres.exe` compiling `resource.rc` -> `resource.rc.obj`
+            target_link_options(cataclysm-tiles PRIVATE
+                "LINKER:--strip-debug"
```
Comment (Qrox) [inline src/CMakeLists.txt:52]: Why is this needed here? Debug symbols are already determined by the build type.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -48,13 +48,13 @@ if (TILES)
                 ${MESSAGES_CPP}
                 ${RESOURCE_RC})
         if(MSYS2)
-            # /MANIFESTINPUT not supported by GNU linker
-            # Defaul embed via `windres.exe` compiling `resource.rc` -> `resource.rc.obj`
+            target_link_options(cataclysm-tiles PRIVATE
+                "$<$<CONFIG:Debug,RelWithDebInfo>:LINKER:--strip-debug>"
```
Comment (Qrox) [inline src/CMakeLists.txt:52]: Maybe I misunderstand but does this remove debug info from the debug build? Having the debug info is the very purpose of the debug build so I doubt this is useful to anybody.

> Need `--strip-debug`, otherwise I get:
> 
> ```
> Program 'cataclysm-tiles.exe' failed to run: The specified executable is not a valid application for this OS plat
> ```

If you're getting this from the debug build, a possible reason is that the executable exceeds 4 GB and cannot be loaded by Windows. In that case you can use the following commands to separate the debug symbol from the executable or remove the debug info in your local workflow, but I don't think we should remove that information in the in-repo script.

```bash
objcopy --only-keep-debug --compress-debug-sections cataclysm-tiles.exe cataclysm-tiles.debug
objcopy --strip-unneeded --add-gnu-debuglink=cataclysm-tiles.debug cataclysm-tiles.exe
```

Comment (alef) [reply src/CMakeLists.txt:52]: Please check #71805
